### DISCUSSION TITLE
Running Kong as a Docker container on Azure App Service

### DISCUSSION THREAD

**Discussion by codebased** [Kong/kong#8472] [Help]
Hello All,

I have a custom docker image for Kong API Gateway. I have enabled MTLS with CA root. When I run the docker on my local machine it works as expected. 

For the release , I am deploying KONG API Gateway on Azure App Service Platform for Linux using a docker image, which is working without MTLS enabled. However what has been observed that by running kong api gateway with MTLS enabled, always returned as 401. I am passing my same certificate that I was passing in my local machine. The CA root is also the same.

According to the kong documentation, it says:

> If Consumer did not present a valid certificate (this includes requests not sent to the HTTPS port), then the response will be HTTP 401 No required TLS certificate was sent
 
After reading this document, it seems like the certificate has been striped before even it reaches the kong :

> TLS termination of the request happens at the frontend load balancer. When forwarding the request to your app code with [client certificates enabled](https://docs.microsoft.com/en-us/azure/app-service/app-service-web-configure-tls-mutual-auth#enable-client-certificates), App Service injects an X-ARR-ClientCert request header with the client certificate.

https://docs.microsoft.com/en-us/azure/app-service/app-service-web-configure-tls-mutual-auth#access-client-certificate 

Does anyone of you have this problem ?

**Comment by flrgh** [comment]
Hi :wave: this question appears to be in reference to the Kong EE [mtls-auth plugin](https://docs.konghq.com/hub/kong-inc/mtls-auth/). We recommend getting in touch with the support team to receive further assistance.
